fontra2fontir/src/source.rs

```diff
@@ -244,24 +247,127 @@ mod tests {
     use super::*;
 
     #[test]
-    fn glyph_names_of_minimal() {
+    fn glyph_info_of_minimal() {
         let mut source = FontraIrSource::new(testdata_dir().join("minimal.fontra")).unwrap();
-        let inputs = source.inputs().unwrap();
+        // populates glyph_info
+        source.inputs().unwrap();
         assert_eq!(
-            vec![GlyphName::new(".notdef")],
-            inputs.glyphs.keys().cloned().collect::<Vec<_>>()
+            Arc::new(
+                vec![(
+                    GlyphName::new(".notdef"),
+                    (
+                        "../resources/testdata/fontra/minimal.fontra/glyphs/%2Enotdef.json".into(),
+                        vec![]
+                    )
+                )]
+                .into_iter()
+                .collect::<BTreeMap<_, _>>()
+            ),
+            source.glyph_info
         );
     }
 
     #[test]
-    fn glyph_names_of_2glyphs() {
+    fn glyph_info_of_2glyphs() {
         let mut source = FontraIrSource::new(testdata_dir().join("2glyphs.fontra")).unwrap();
-        let inputs = source.inputs().unwrap();
-        let mut glyph_names = inputs.glyphs.keys().cloned().collect::<Vec<_>>();
-        glyph_names.sort();
+        // populates glyph_info
+        source.inputs().unwrap();
         assert_eq!(
-            vec![GlyphName::new(".notdef"), GlyphName::new("u20089")],
-            glyph_names
+            Arc::new(
+                [
+                    (
+                        GlyphName::new(".notdef"),
+                        (
+                            "../resources/testdata/fontra/2glyphs.fontra/glyphs/%2Enotdef.json"
+                                .into(),
+                            vec![]
+                        )
+                    ),
+                    (
+                        GlyphName::new("u20089"),
+                        (
+                            "../resources/testdata/fontra/2glyphs.fontra/glyphs/u20089.json".into(),
+                            vec![0x20089]
+                        )
+                    )
+                ]
+                .into_iter()
+                .collect::<BTreeMap<_, _>>()
+            ),
+            source.glyph_info
         );
     }
+
+    #[test]
+    fn glyph_info_of_mutator_sans() {
```

Comment:
The name of the test isn't tremendously informative wrt what it validates. Based on the PR title the goal is to confirm support for 0..N codepoints mapped to a glyph. I suggest validating that a bit more minimally.

For example, rather than adding the entire MutatorSans and checking tons of glyphs add a new test font that has the absolute minimum to verify what you wish to test. Perhaps a new manually created font that has say .notdef (no codepoints), A (1 codepoint), Aaacute (2 codepoints) and some emoji that has lots (say https://emojipedia.org/handshake-medium-light-skin-tone-medium-skin-tone#technical).

Reply:
https://github.com/googlefonts/fontc/pull/754 (PR to this branch) aspires to so update